### PR TITLE
switch download URLs based on absetup

### PR DESCRIPTION
**Pull request by scrajesh**
1. `absetup=expB` would send secret download URL with the proxy cached header
2. default value is `expA` i.e. URL with no cache header
3. new package `electrode-ota-server-ccm` requires either dao-app implementation or override in public-instance
4. using the above package, updateCheck would returns download info with/without cached header

### CODE REVIEW

File: electrode-ota-server-routes-acquisition/src/index.js [hunk 1]
```diff
@@ -32,6 +32,11 @@ export const register = diregister({
                 console.log("error making update check ", request.query, e.message);
                 return reply(e);
             }
+            // default is Experiment-A(expA) => Non-cached dowload URL
+            // Experiment-B(expB) => Cached dowload URL (/foo_getp)
+            if (qs.absetup === "expB" && "downloadURL" in updatedInfo) {
+                updatedInfo.downloadURL = updatedInfo.downloadURL.replace("storagev2", "foo_getp");
```
Comment (datvong-wm) [inline electrode-ota-server-routes-acquisition/src/index.js:38]: let's not call this foo-getp

Comment (scrajesh) [reply electrode-ota-server-routes-acquisition/src/index.js]: now it's `/deltaPackage`

File: electrode-ota-server-ccm/src/ccm-store.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// includes
+const { fetchJSON } = require("@walmart/electrode-fetch");
```
Comment (datvong-wm) [inline electrode-ota-server-ccm/src/ccm-store.js:2]: can't use this

File: electrode-ota-server-ccm/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "1.0.0",
+  "main": "lib/index.js",
+  "scripts": {
+    "test": "ota-mocha",
+    "build": "ota-babel",
+    "coverage": "ota-nyc",
+    "prepublish": "npm run build"
+  },
+  "dependencies": {
+    "@walmart/electrode-ccm-store": "^1.0.0",
```
Comment (datvong-wm) [inline electrode-ota-server-ccm/package.json:12]: can't access this

Comment (scrajesh) [reply electrode-ota-server-ccm/package.json:12]: but OTA-Public-instace does include private fileservice package, ain't it?

Comment (datvong-wm) [reply electrode-ota-server-ccm/package.json:12]: they put it there and not here.

Comment (scrajesh) [reply electrode-ota-server-ccm/package.json]: moved it to our public instance

File: electrode-ota-server-diregister/src/index.js [hunk 4]
```diff
@@ -33,16 +33,16 @@ const result = (obj, dep) => {
 export const HANDLERS = {
     electrode: (server, dep, options) => result(server, dep)
 };
-const resolveable = (name) => {
+const resolveable = name => {
     const ret = {name};
-    ret.promise = new Promise(function (_resolve, _reject) {
+    ret.promise = new Promise(((_resolve, _reject) => {
```
Comment (datvong-wm) [inline electrode-ota-server-diregister/src/index.js:38]: too many ()?

Comment (scrajesh) [reply electrode-ota-server-diregister/src/index.js:38]: eslint fixes. each time i work with a ticket, trying to fix them one file at time.

Comment (datvong-wm) [reply electrode-ota-server-diregister/src/index.js:38]: there seems to be an extra ()

Comment (scrajesh) [reply electrode-ota-server-diregister/src/index.js]: removed

File: electrode-ota-server-ccm/src/ccm-store.js (hunk 2 above)
Comment (scrajesh) [reply electrode-ota-server-ccm/src/ccm-store.js]: moved to our private package

**Comment by scrajesh** [comment]
> add tests

splitted one big file of routes acquisition into multiple files to add unit tests.
unit tests for a/b test have been added.

**Comment by scrajesh** [comment]
Closing this PR due to beta release happened in the same branch.

It's moved to: https://github.com/electrode-io/electrode-ota-server/pull/130

**Pull request closed without merging**